pymongo/mongo_client.py

```diff
@@ -1600,9 +1605,14 @@ def _process_kill_cursors(self):
     def _process_periodic_tasks(self):
         """Process any pending kill cursors requests and
         maintain connection pool parameters."""
-        self._process_kill_cursors()
         try:
+            self._process_kill_cursors()
             self._topology.update_pool(self.__all_credentials)
+        except InvalidOperation:
+            if self._topology._closed:
+                pass
+            else:
+                helpers._handle_exception()
         except Exception:
```

Comment:
This might be cleaner:
```python
except Exception as exc:
    if isinstance(exc, InvalidOperation) and self._topology._closed:
        return # Ignore when client has been closed. 
```

Reply:
Done.